### ISSUE TITLE
How can we revive Quartz ?

### ISSUE THREAD

**Issue by jarlah** [quartz-scheduler/quartz#277]
The plan is to revive this project. Can you help? Is there any value of doing so? Can we fix the issues necessary to make quartz usable again?

I previously cloned the repo but has since deleted it. 

The discussion below will bring light to what we need to do.

**Comment by jarlah** [comment]
Decoupled it completely so the project is now not a fork, but a clone

**Comment by jarlah** [comment]
i have now ripped out everything else than core logic. there is only three modules left, core, jobs and plugins. The plan is to build the ecosystem from ground up, including deployment, code quality checks ++. If someone wonder why there is only three modules there, well, if you are using jboss or weblogic, maybe we could try to make this new clone class compatible with quartz-weblogic and quartz-jboss etc. No need to live in the pre historic days of container managed scheduling systems.

**Comment by jarlah** [comment]
Travis builds, jitpack is working, and I think that the new clone can be easily used together with the weblog, jboss, oracle, stub stuff from this project. I will make an example app for it. I have added a simple helloworld example app here https://github.com/OpenQuartz/quartz-helloworld which uses jitpack to download dependencies.

**Comment by jarlah** [comment]
added example on how to use the new clone with the old jboss package from this project:
https://github.com/OpenQuartz/quartz-hello-j2ee
I have no idea what i am playing with, but it runs ;)

**Comment by jarlah** [comment]
I have the following comment about people clinging to this dead project:

https://goo.gl/images/fTCCp4

Come over to my project instead.

**Comment by zemian** [comment]
Hello @jarlah 

It's good to see you take interests in Quartz dev! I admit that I have not been able to contribute much in the past couple of years. And unfortunately I don't see much activity from Terracotta either :(

I have contributed time in this project in the past and even created few supporting projects based on this. So myself would definitely want to see it pickup activity again and continue to add new features.

However I am not sure making a clone is a good start at this point. Have you sent emails to mailing groups or voice what major issues you currently having that's needed a clone? What issues have fixed in clone that's not in here? There is a proposed Quartz3 wiki that requesting inputs as what you like to see as well. I would like to give respect to old team/project and seek advice before forking the project.

There are several issues that I would like Terracotta folks to give more directions such as removing their specific code in modules, or make it an optional at least. I used to talk to @jhouserizer on these matters, but he has been quiet here lately. Hopefully he can comment here as well.

**Comment by jarlah** [comment]
Its all good :) Nice to see that my actions have triggered you to reply :) if there is plans or possibilities i am more than happy to delete my clone.

So i just want to get something happening. I contacted terracotta devs and got this respons

"Hello Jarl
With Quartz Scheduler being fully Open Sourced, we do not have a plan to release an update in the immediate future.

We understand the desire of the community to see an updated release version, and would hope that a more active fork can be created and maintained outside of our control.

Fiona O’Shea
IoT Analytics Terracotta Dev West“

**Comment by zemian** [comment]
@jarlah 

Ah, seeing that reply from Terracotta, I can understand your actions. Note that I am mere a contributor, not the owner of this repo either. But to be fair and for respect sake, I would propose the following:

* Quartz has a long history of code and used in many other communities, so it's my take that we should pay as much as respect to these eco systems as possible. 
* From my experience is that maintaining a healthy open project is not easy and it's very time consuming. For this reason, we ought to play nice and take as much help as we can with community input. Things need to be open and have a healthy discussion first.
* As for immediate todo actions, we should identify most glaring issues in this original repo. I have few issues here of my own that I like to see resolve as well:
  ** What are the current major fixes in release 2.3.x that are completed and waiting on a release? What does the community want to see in this release?
  ** We have a Quartz3 wiki that requesting community for future roadmap and feature requests. I don't see much activity there. So we should fill that out so we can see future.
  ** Be able to perform a public release easier and at anytime. I think Terracotta is sponsoring this project, so they are providing their infrastructure to build and release as well. Unfortunately they are not very active in this area lately.
  ** The project has few terracotta specific code/modules that I don't feel it should be here as part of the Quartz core.
  ** I think creating/moving the documentation into GitHub wiki is easier to maintain than as separate static web site.
  ** Contribution code should be more easier, rather than forced to sign consent letter from Terracotta. I am not a lawyer, but I can see this step prevent people from contributing. Either we make it easier, or remove it.
* Only after we attempted above and no responses from original authors, then I would fee comfortable forking the repo.

**Comment by jarlah** [comment]
Agree. 

We need however to formulate a plan of relevant actions to reach the goal you mentioned. 

I am in a limbo here since i am not a member of the repo. So i cannot contribute on wiki. So i would suggest using Trello or similar solution to list, group and manage todo items for such an effort. A wiki is not dynamic enough. But it can be the end goal.

I have removed all terracotta specific and hard to port code in my clone. I have trimmed it down to an absolute minimum. This enables me to improve the code. And use it. I can however stop the marketing of this clone and remove the ported issues and delete comments. But i will not remove the clone yet. Maybe if i see some progress. 

I am a senior developer and architect that contribute. But my actual end goal is just to revive the project.

**Comment by zemian** [comment]
I think GitHub already brings contribution tools easy already, if one is willing to contributing that is, there is no limit. For example, open an Issue subject: Quartz3 feature requests, and start writing whatever you want to discuss. No permission is needed. Pull request can be created without permission either. If desired is strong, next is submit to be a contributor and get access to wiki and a code committer. We can start with those right?

Live Project needs Active Developers! It doesn't t help much if we revived it with only surface tools such as Trello, or a simple fork repo, but we don't have developers to actually maintain the code.

**Comment by jarlah** [comment]
No problem accepting your comments. 

I will clean up this repo and remove irrelevant links and comments later today. 

Then we can start to plan what to do with github issues

**Comment by jarlah** [comment]
Closing this issue. It can be continued and discussed anyhow.

**Comment by zemian** [comment]
Thanks @jarlah !

Please do contribute by helping. You can help answer questions/issues, or review PR. See http://www.quartz-scheduler.org/community/ for more info on how to help. If you feel any portion of code that can be improve, provide the PR, and I will try to review when I can. Any improvement to project development process can be send to DEV LIST on as well.

**Comment by jarlah** [comment]
btw @zemian, I have poked around on your profile and homepage, and see that you are heavily invested in quartz. Its a very nice framework, I give you that. I have some problems finding a one to one communication channel, so it must be in this issue. I have a private testing bench for playing around with quartz, dependencies and modules, at https://github.com/OpenQuartz/scheduler. This is however a different repo I linked to previously and is very clearly marked as unofficial. My point wit this is that it shows how I feel would be better suited for an ongoing join effort between open source developers like you and me. There is a fundamental problem when one or two persons sitting on the private keys necessary to deploy at maven central, is the main gate to get things rolled out. So in my unofficial clone I have used jitpck as artifact hosting The only thing needed to get things rolled out is to make a tag. That simple. I have always felt maven central deployment to be a major pain. Nothing can change that.

My main objective, with having something like this to play with (my first attempt was not like that), is to find out where there are things to improve in the code. Until now I have not found a single thing to improve directly. I have looked for new Thread references, but only found aprox one (shutdown). I have also poked around for bad patterns. None directly interesting. So what I see is a project that need bugfixes, and improvements. When no new development is done, projects like these tend to be quite stable and that's good. No problem there :+1: 

But anyway, I have a feeling that the end result is a different repo from this one. So I am just giving you a heads up, that I have my own private test bench where I try different things, remove some things but yet trying to keep then overall structure as identical to the original as possible. If we **do** get access to the repo, my test bench can be a proof of concept on how we can restructure the repo. But don't see my jitpack comment as a "I think we should". I am a pragmatic senior lead developer that finds the easiest solution to be the best in most cases ;)

**Comment by zemian** [comment]
@jarlah I actually haven't heard of https://jitpack.io/ before, I will check it out. I don't handle the Quartz release so I don't know their process. I posted to DEV LIST about this topic, but I am not sure anyone is responding :(

If you checkout the Quartz3 todo list, I have updated with things I like to add/work on. Would be nice to explore those.

To me, the release process is just part of the dev cycle. I vote for whatever that's easiest to maintain. I also support those who is willing to spent time in it and keep it up. Maven central might not be easiest to deal with, but if someone willing to do it, then I don't see a problem. Or like you say if there is alternative, we can explore that too. But that will require the project owner to approve obviously. I am not sure what Terracotta's interests in this project at this point, since it's been neglected for a while now. I also posted a email to DEV LIST. Let's wait a little and see.

**Comment by HipDragon** [comment]
Another possibility is to align with https://www.quartz-scheduler.net/. This would allow us to make sure that this library doesn't fall behind in features. Any word on ways to keep this project alive and moving forward? I do think that we need to keep this in the Maven central.

**Comment by jarlah** [comment]
i haven't got to work on this lately, but i still have my unofficial clone at https://github.com/OpenQuartz/scheduler. I have signed developer agreement with terracotta, so I guess the legal things is OK for going forward with anything at all with a clone, fork or whatever else might be a solution. I think fork is not a good idea, because there will never be upstream pushes. Hence the clone.

@zemian i see that your appeal for suggestions for things to add to the quartz3 roadmap has got no replies on the mailing list. Actually when, hen i checked the mailing list, I was actually surprised to see someone who still believes the project is active. Then i saw that it was you :) I also saw a question not too far down the list where a person asks if this project is alive.

So conclusively, if after a month no one has replied for such an appeal, I would strongly recommend to plan another course of action than to try to fill this roadmap and convince terracotta to open up the repo.

I have no special agenda in this, if the project is revived, great, then I can use it in my own projects and know that its maintained. Otherwise, I will continue to kick this project in the right direction by making suggestions :)
